src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -221,6 +227,48 @@ public override string EditFormatString
         }
 
         /// <inheritdoc />
+        public override ModelMetadata ElementMetadata
+        {
+            get
+            {
+                if (!_haveCalculatedElementMetadata)
+                {
+                    // Short-circuit checks below. As in IsCollectionType, do not consider string a collection.
+                    _haveCalculatedElementMetadata = ModelType == typeof(string);
+                }
```

Comment:
> Why not check IsCollectionType here?

Because that repeats many of the checks needed below and only tells us `ElementMetadata` should be non-`null`. IOW its repetitive and not that helpful.